### ISSUE TITLE
NPE Crash with malformed XML config

### ISSUE THREAD

**Issue by AndrewPWRE** [opendcs/opendcs#753]
**Describe the bug**
Dump:

```
.\7013-RC09\bin\rledit.bat -P .\decodes.properties -d 3
Exception in thread "main" java.lang.NullPointerException
        at decodes.rledit.DTEquivTableModel.rebuild(DTEquivTableModel.java:41)
        at decodes.rledit.DTEquivTableModel.<init>(DTEquivTableModel.java:30)
        at decodes.rledit.RefListFrame.<init>(RefListFrame.java:91)
        at decodes.rledit.RefListEditor.<init>(RefListEditor.java:25)
        at decodes.rledit.RefListEditor.main(RefListEditor.java:113)
```

-d 3 log:
```
INFO    10/24/24 03:39:47 Set DecodesSettings source=.\decodes.properties
INFO    10/24/24 03:39:47 Set DecodesSettings source=.\decodes.properties
DBG1    10/24/24 03:39:47 Forwarding javax.logging to ilex log.
INFO    10/24/24 03:39:47 After parseArgs, DecodesSettings src file=.\decodes.properties
INFO    10/24/24 03:39:47 RefListEditor Starting (OPENDCS 7.0 built on 7.0.13) =====================
DBG1    10/24/24 03:39:47 Creating XmlDatabaseIO for directory 'c:/users/agilmore/'
```

I think this XML database directory of my user doesn't have any of the expected subdirs/files in it.

**Expected behavior**
Error message about unable to locate the XML database.

**Desktop/ServerSide (please complete the following information):**
 - OS: Windows 10
 - Version 7.0.13-RC09

**Comment by MikeNeilson** [comment]
Hmm. A default setup should have those available. Though rledit should
definitely be the one app that doesn't crash if one of the sets is missing
since it's for creating them.

**Comment by MikeNeilson** [comment]
`.\7013-RC09\bin\rledit.bat -P .\decodes.properties -d 3`
                            ^-------------^ you also shouldn't be using a properties file in the bin directory. 

default install with no DCSTOOL_USERDIR in the environment will automatically create ~/.opendcs (<user>\AppData\Roaming\.opendcs on windows) and copy the baseline xml database and make a user.properties there.

following that i could recreate in ubuntu, but will try windows tomorrow just to verify.

**Comment by MikeNeilson** [comment]
okay, wasn't using 
"install" but using a fresh `DCSTOOL_USERDIR` the base database was created correctly and no NPE on rledit.

but now that I read your comment more awake perhaps you intentionally misconfigured, in which case yeah the error messaging isn't create, but without a correct edit-db directory the whole operation should fail much earlier.

That is one of the checks I'm adding in the new "everything from a DataSource" mechanism so while valid, I don't think this should stop 7.0.13 as that's likely been the behavior for a while.

**Comment by agilmore2** [comment]
It wasn't an intentional misconfiguration, but one I ended up with somehow.
That decodes.properties wasn't created recently at least.

**Comment by msweier** [comment]
I was messing around defining cwms parameters using rledit based on discussion in https://github.com/opendcs/opendcs/issues/749 and my human-readable enumerator was wiped. Using desktop client 7.0.12. Maybe related to this bug? Had to change my output for cwms routing because it wouldn't load with the human-readable output defined.

**Comment by ktarbet** [comment]
I have reproduced the issue.  Here is what I have learned.

If the directories don't exist the XmlDatabaseIO will create them all, but they are now empty, leading to later error that is more challanging to report properly.
 
https://github.com/opendcs/opendcs/blob/a6e30e4597641b10ad07125fc58f1709192b07b0/java/opendcs/src/main/java/decodes/xml/XmlDatabaseIO.java#L161

Failing in the XmlDatabaseIO constructor would be easier , but  do we need to support creating an empty XML database?   Perhaps another argument that indicates existing Database versus empty?
